dynawo/sources/Models/CPP/ModelNetwork/DYNModelNetwork.cpp

```diff
@@ -192,20 +192,20 @@ ModelNetwork::initializeFromData(const shared_ptr<DataInterface>& data) {
     // =============================
     //    CREATE SWITCH MODEL
     // =============================
-    const vector<shared_ptr<SwitchInterface> >& switches = (*iVL)->getSwitches();
-    vector<shared_ptr<SwitchInterface> >::const_iterator iSwitch;
+    const vector<std::shared_ptr<SwitchInterface> >& switches = (*iVL)->getSwitches();
+    vector<std::shared_ptr<SwitchInterface> >::const_iterator iSwitch;
     for (iSwitch = switches.begin(); iSwitch != switches.end(); ++iSwitch) {
       string id = (*iSwitch)->getID();
       componentsById[id] = (*iSwitch);
-      shared_ptr<ModelSwitch> modelSwitch(new ModelSwitch(*iSwitch));
+      std::shared_ptr<ModelSwitch> modelSwitch = ModelSwitchFactory::newInstance(*iSwitch);
```

Comment:
Why creating a factory here and not for the others?

Reply:
Using a factory makes sure we build ModelSwitch instances with a std::shared_ptr. The ModelSwitch class can't be used with std::unique_ptr because it inherits from std::enable_shared_from_this<ModelSwitch>.

Reply:
ok, but why only the switch? others components are still created locally in the method with no factory involved